### DISCUSSION TITLE
Possible to use changie in a github action for pull requests?

### DISCUSSION THREAD

**Discussion by anrico-oltmanns** [miniscruff/changie#436] [Q&A] [answered]
In our organization, we merge everything to "develop". Once a week or bi-weekly we merge everything from "develop" to "master". We want to have a list of all the issues associated with the PR in the PR description based on the commits, which are linked to the issues (with #issue-number).

So basically, I want to have the change logs from changie in our master PR description. Is it possible? Ive tried a lot of things but nothing seemed to work. Also, I couldnt find a workaround for our specific case yet.

**Comment by miniscruff** [comment, marked as answer]
I have built a GitHub action that should get you started: https://github.com/miniscruff/changie-action
Changie itself has a GitHub workflow that creates a PR for releases here: https://github.com/miniscruff/changie/blob/main/.github/workflows/gen-release-pr.yml just replace `go run main.go` with the action above.
You can use the newly generated version file as the body of the PR as well.

**Reply by anrico-oltmanns** [reply]
The problem with this is that I dont want to have it as a `workflow_dispatch`. Also, I dont want to have any inputs. I just want to add a list of the issues in the master PR we do every week. It seems like this specific github action is configured that way that you have to have an input. Im pretty new to github actions and I couldnt get it to work so far.

**Reply by anrico-oltmanns** [reply]
Also, do I simply have to replace `go run main.go` with `go run .github/workflows/changie.yml` if I named it like that? Or how do I do it? I tried it and it didnt seem to work

**Reply by anrico-oltmanns** [reply]
Maybe Im missing the fundamentals. When using the github action you mentioned [here](https://changie.dev/guide/installation/) I Already get the error message `open .changie.yml: no such file or directory` + `Error: The process '/opt/hostedtoolcache/changie-action/1.10.2/x64/changie' failed with exit code 1`
![Screenshot 2023-01-24 124615](https://user-images.githubusercontent.com/114996994/214283725-fed2f4c9-fc7f-465e-9236-89b66b8cb0f1.png)

and yes, im sure that the PAT im using has every required permission.

**Reply by miniscruff** [reply]
The error you posted is due to a missing config file changie needs to run. Are you using changie to manage your changelogs currently?

**Reply by miniscruff** [reply]
I may have misunderstood the original question, are you asking to use changie purely for this one PR and not using changie in each PR/change? If so, no changie will not work that way. You would have to use changie for every change in order to generate the fragments needed to build the release notes, changelog and PR description. It will not work only on PR numbers or anything like that.

**Reply by anrico-oltmanns** [reply]
Alright. Yea, that was my original question. Just using it for this one PR. Thank you for your help. Ill look for another tool then 👍

**Reply by miniscruff** [reply]
You might be able to use the github release note autogeneration but not sure.
